packages/office-ui-fabric-react/src/components/Nav/Nav.scss

```diff
@@ -96,6 +96,11 @@ $linkRightPadding: 20px;
   vertical-align: middle;
 }
 
+.ms-Nav-linkHorizontal {
+  @include float(left);
+  display: inline;
```

Comment:
in the debug session, seems display: inline alone is not enough.  As for allow consumer to override the class, we can if need is there.  Currently with @include float(left) it should take care of local LTR or RTL issue.